### PR TITLE
OP-1129 OP-1142 Script language update

### PR DESCRIPTION
**Pull request by mizzioisf**
Update: should be superseded by https://github.com/informatici/openhospital-gui/pull/1959

See OP-1129 an OP-1142.

- Updated language list supported in scripts
- Better handling of language selection for user

### CODE REVIEW

File: oh.ps1 [hunk 1]
```diff
@@ -98,14 +98,15 @@ $global:ProgressPreference= 'SilentlyContinue'
...
-# OH_PATH="c:\Users\OH\OpenHospital\oh-1.13"
+# OH_PATH="c:\Users\OH\OpenHospital\oh-1.14"
 
 # set OH mode to PORTABLE | CLIENT | SERVER - default set to PORTABLE
 #$script:OH_MODE="PORTABLE"
 
 # language setting - default set to en
-$script:OH_LANGUAGE_LIST="en|fr|es|it|pt|ar"
-$script:OH_LANGUAGE="en" # default
+$script:OH_LANGUAGE_LIST= @("al", "ar", "de", "en","es","fr","it","pt")
```
Comment (mwithi) [inline oh.ps1:107]:
```suggestion
$script:OH_LANGUAGE_LIST= @("al","ar","de","en","es","fr","it","pt")
```

File: oh.ps1 [hunk 2]
```diff
@@ -98,14 +98,15 @@ $global:ProgressPreference= 'SilentlyContinue'
 ##################### OH general configuration ####################
 
 # -> OH_PATH is the directory where Open Hospital files are located
-# OH_PATH="c:\Users\OH\OpenHospital\oh-1.13"
+# OH_PATH="c:\Users\OH\OpenHospital\oh-1.14"
```
Comment (mwithi) [inline oh.ps1:101]: Reverting
```suggestion
# OH_PATH="c:\Users\OH\OpenHospital\oh-1.13"
```

File: oh.sh [hunk 3]
```diff
@@ -43,14 +43,14 @@ WRITE_CONFIG_FILES="off"
 ############## OH general configuration - change at your own risk :-) ##############
 
 # OH_PATH is the directory where Open Hospital files are located
-# OH_PATH=/usr/local/OpenHospital/oh-1.13
+# OH_PATH=/usr/local/OpenHospital/oh-1.14
```
Comment (mwithi) [inline oh.sh:46]: Reverting
```suggestion
# OH_PATH=/usr/local/OpenHospital/oh-1.13
```

File: oh.sh [hunk 4]
```diff
@@ -521,11 +528,11 @@ desktop_path=$(xdg-user-dir DESKTOP)
 echo "[Desktop Entry]
 	Type=Application
 	# The version of the Desktop Entry Specification
-	Version=1.13.0
+	Version=1.14.0
```
Comment (mwithi) [inline oh.sh:531]:
```suggestion
	Version=1.13.0
```

File: oh.sh [hunk 5]
```diff
@@ -521,11 +528,11 @@ desktop_path=$(xdg-user-dir DESKTOP)
...
 	# The name of the application
 	Name=OpenHospital
 	# A comment which will be used as a tooltip
-	Comment=Open Hospital 1.13 shortcut
+	Comment=Open Hospital 1.14 shortcut
```
Comment (mwithi) [inline oh.sh:535]: Reverting
```suggestion
	Comment=Open Hospital 1.13 shortcut
```

File: oh.ps1 [hunk 6]
```diff
@@ -104,8 +104,9 @@ $global:ProgressPreference= 'SilentlyContinue'
 #$script:OH_MODE="PORTABLE"
 
 # language setting - default set to en
-$script:OH_LANGUAGE_LIST="en|fr|es|it|pt|ar"
-$script:OH_LANGUAGE="en" # default
+$script:OH_LANGUAGE_LIST= @("al","ar","de","en","es","fr","it","pt")
```
Comment (mwithi) [inline oh.ps1:107]:
```suggestion
$script:OH_LANGUAGE_LIST= @("ar","de","en","es","fr","it","pt","sq")
```

File: oh.sh [hunk 7]
```diff
@@ -49,8 +49,8 @@ WRITE_CONFIG_FILES="off"
 #OH_MODE="PORTABLE" 
 
 # language setting - default set to en
-OH_LANGUAGE_LIST="en|fr|es|it|pt|ar"
-#OH_LANGUAGE="en" # default
+OH_LANGUAGE_LIST=("al" "ar" "de" "en" "es" "fr" "it" "pt")
```
Comment (mwithi) [inline oh.sh:52]:
```suggestion
OH_LANGUAGE_LIST=("ar" "de" "en" "es" "fr" "it" "pt" "al")
```

or (to not switch INFO)

```suggestion
OH_LANGUAGE_LIST=("sq" "ar" "de" "en" "es" "fr" "it" "pt")
```

Comment (mizzioisf) [reply oh.sh]: fixed, thanx

File: oh.sh [hunk 8]
```diff
@@ -213,7 +213,7 @@ function script_menu {
 	echo "   -C    set OH in CLIENT mode"
 	echo "   -P    set OH in PORTABLE mode"
 	echo "   -S    set OH in SERVER mode (portable)"
-	echo "   -l    [ $OH_LANGUAGE_LIST ] -> set language"
+	echo "   -l    set language -> [ ${OH_LANGUAGE_LIST[*]} ]"
```
Comment (mwithi) [inline oh.sh:216]:
```suggestion
	echo "   -l    set OH language -> [ ${OH_LANGUAGE_LIST[*]} ]"
```

File: oh.sh [hunk 9]
```diff
@@ -43,14 +43,14 @@ WRITE_CONFIG_FILES="off"
...
+# OH_PATH=/usr/local/OpenHospital/oh
 
 # set OH mode to PORTABLE | CLIENT | SERVER - default set to PORTABLE
 #OH_MODE="PORTABLE" 
 
 # language setting - default set to en
-OH_LANGUAGE_LIST="en|fr|es|it|pt|ar"
-#OH_LANGUAGE="en" # default
+OH_LANGUAGE_LIST=("ar" "de" "en" "es" "fr" "it" "pt" "sq")
+OH_LANGUAGE_LIST_INFO=("arabic" "german" "english" "spanish" "french" "italian" "portuguese" "albanian")
```
Comment (mwithi) [inline oh.sh:53]: should we capitalize them?

```suggestion
OH_LANGUAGE_LIST_INFO=("Arabic" "German" "English" "Spanish" "French" "Italian" "Portuguese" "Albanian")
```

File: oh.ps1 [hunk 10]
```diff
@@ -98,14 +98,15 @@ $global:ProgressPreference= 'SilentlyContinue'
...
+# OH_PATH="c:\Users\OH\OpenHospital\oh"
 
 # set OH mode to PORTABLE | CLIENT | SERVER - default set to PORTABLE
 #$script:OH_MODE="PORTABLE"
 
 # language setting - default set to en
-$script:OH_LANGUAGE_LIST="en|fr|es|it|pt|ar"
-$script:OH_LANGUAGE="en" # default
+$script:OH_LANGUAGE_LIST= @("ar","de","en","es","fr","it","pt","sq")
+$script:OH_LANGUAGE_LIST_INFO=("arabic","german","english","spanish","french","italian","portuguese","albanian")
```
Comment (dbmalkovsky) [inline oh.ps1:108]: same comment as there is for line 53 in oh.sh below

Comment (mizzioisf) [reply oh.ps1:108]: above ? :-) :-)

Comment (mizzioisf) [reply oh.ps1:108]: See https://github.com/informatici/openhospital-gui/pull/1959

File: oh.sh (hunk 9 above)
Comment (mizzioisf) [reply oh.sh:53]: See https://github.com/informatici/openhospital-gui/pull/1959

**Comment by mwithi** [comment]
Superseded by https://github.com/informatici/openhospital-gui/pull/1959

**Pull request closed without merging**